service/src/main/java/com/codeforcommunity/requester/TreeBenefitsCalculator.java

```diff
@@ -0,0 +1,189 @@
+package com.codeforcommunity.requester;
+
+import static org.jooq.generated.Tables.TREE_BENEFITS;
+import static org.jooq.generated.Tables.TREE_SPECIES;
+
+import com.codeforcommunity.exceptions.ResourceDoesNotExistException;
+import java.util.HashMap;
+import java.util.Map;
+import org.jooq.DSLContext;
+import org.jooq.Field;
+import org.jooq.generated.tables.records.TreeSpeciesRecord;
+
+public class TreeBenefitsCalculator {
+  private final DSLContext db;
+  final String speciesCode;
+  final double diameterCM;
+  static final double[] intervals =
+      new double[] {3.81, 11.43, 22.86, 38.10, 53.34, 68.58, 83.82, 99.06, 114.30};
+  final double x0;
+  final double x1;
+  static final Map<String, Double> currencyConversion =
+      new HashMap<String, Double>() {
+        {
+          put("electricity_kwh_to_currency", 0.1401);
+          put("natural_gas_kbtu_to_currency", 0.01408);
+          put("h20_gal_to_currency", 0.0008);
+          put("co2_lb_to_currency", 0.00334);
+          put("o3_lb_to_currency", 4.59);
+          put("nox_lb_to_currency", 4.59);
+          put("pm10_lb_to_currency", 8.31);
+          put("sox_lb_to_currency", 3.48);
+          put("voc_lb_to_currency", 2.31);
+        }
+      };
+  static final Map<String, Double> unitConversion =
+          new HashMap<String, Double>() {
+            {
+              put("kBTU_to_kWh", 0.2930);
+              put("kg_to_lb", 2.20462);
+              put("m3_to_gal", 264.172);
+            }
+          };
+
+  // diameter in inches
+  public TreeBenefitsCalculator(DSLContext db, String commonName, double diameter) {
+    this.db = db;
+    this.speciesCode = getSpeciesCode(commonName);
+    this.diameterCM = diameter * 2.54;
+
+    double x0 = 0;
+    double x1 = 0;
+    for (int i = 0; i < intervals.length; i++) {
+      if (intervals[i] <= diameterCM) {
+        x0 = intervals[i];
+        x1 = intervals[i + 1];
+      }
+    }
+    if (x0 == 0 || x1 == 0) {
+      throw new IllegalArgumentException("diameter must be non-zero");
+    }
+    this.x0 = x0;
+    this.x1 = x1;
```

Comment:
Thanks for pointing that out. I didn't actually see their implementation before. When we have to clamp, that means interpolation will break (division by 0 since x0=x1), so I will just return y0 (or y1) whenever x0==x1